src/NServiceBus.Core/Features/FeatureComponent.cs

```diff
@@ -15,11 +14,11 @@ public FeatureComponent(SettingsHolder settings)
             this.settings = settings;
         }
 
-        public void RegisterFeatureEnabledStatusInSettings(List<Type> concreteTypes)
+        public void RegisterFeatureEnabledStatusInSettings(HostingComponent hostingComponent)
         {
             featureActivator = new FeatureActivator(settings);
 
-            foreach (var type in concreteTypes.Where(t => IsFeature(t)))
+            foreach (var type in hostingComponent.AvailableTypes.Where(t => IsFeature(t)))
```

Comment:
This is the change that will be needed in https://github.com/Particular/NServiceBus/pull/5498.
In case this PR ends up being merged before #5498 is, I could incorporate the change. Unless there are many other places and a separate PR would be more appropriate.

Reply:
I'll sort this out